docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -269,10 +265,11 @@ public Node call() throws Exception {
                                     //
                                     // deferring the completion of provisioning until the launch
```

Comment:
@KostyaSha the back story is that I have been telling @kohsuke that the Cloud API is "a pile of shite" and he asks for "why" and "where is the evidence". Well this is one of the pieces of evidence: that implementations end up doing this...

The Javadocs clearly state that implementations should not add the node to Jenkins and it is the responsibility of NodeProvisioner to do the adding. Even [@kohsuke has done this](https://github.com/jenkinsci/mansion-cloud-plugin/blob/master/src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/PlannedMansionSlave.java#L247) in that which I previously considered to be the second best cloud api implementation...

A good cloud API would give you the abstractions such that you can get the sequencing of all these events correct and return the node instance at the correct point in time.